objects/src/testing/account_component.rs

```diff
@@ -0,0 +1,107 @@
+use alloc::{sync::Arc, vec::Vec};
+use std::string::ToString;
+
+use assembly::{ast::Module, Assembler, LibraryPath};
+use miden_crypto::dsa::rpo_falcon512::PublicKey;
+
+use crate::{
+    accounts::{AccountComponent, AssembledAccountComponent, StorageSlot},
+    testing::account_code::MOCK_ACCOUNT_CODE,
+    AccountError,
+};
+
+// ACCOUNT COMPONENT ASSEMBLY CODE
+// ================================================================================================
+
+pub const BASIC_WALLET_CODE: &str = "
+    export.::miden::contracts::wallets::basic::receive_asset
+    export.::miden::contracts::wallets::basic::create_note
+    export.::miden::contracts::wallets::basic::move_asset_to_note
+";
+
+pub const RPO_FALCON_AUTH_CODE: &str = "
+    export.::miden::contracts::auth::basic::auth_tx_rpo_falcon512
+";
+
+// BASIC WALLET ACCOUNT COMPONENT
+// ================================================================================================
+
+pub struct BasicWallet;
+
+impl AccountComponent for BasicWallet {
+    fn assemble_component(
+        self,
+        assembler: Assembler,
+    ) -> Result<AssembledAccountComponent, AccountError> {
+        AssembledAccountComponent::compile(BASIC_WALLET_CODE, assembler, vec![])
+    }
+}
+
+// RPO FALCON 512 AUTH ACCOUNT COMPONENT
+// ================================================================================================
+
+pub struct RpoFalcon512 {
+    public_key: PublicKey,
+}
+
+impl RpoFalcon512 {
+    pub fn new(public_key: PublicKey) -> Self {
+        Self { public_key }
+    }
+}
+
+impl AccountComponent for RpoFalcon512 {
+    fn assemble_component(
+        self,
+        assembler: Assembler,
+    ) -> Result<AssembledAccountComponent, AccountError> {
+        AssembledAccountComponent::compile(
+            RPO_FALCON_AUTH_CODE,
+            assembler,
+            vec![StorageSlot::Value(self.public_key.into())],
+        )
+    }
+}
```

Comment:
As mentioned in one of the previous comments, these should not be under the `testing` code. They should probably be in `miden-lib` and ideally, not require assembly at runtime.